### ISSUE TITLE
[META] Purpose of BrowserHTML

### ISSUE THREAD

**Issue by denysvitali** [browserhtml/browserhtml#1303]
I'm opening this issue in order to have some explanations about this project, I don't want to sound like an hater or something: I really enjoy the efforts you're making to do this.
  
I like the fact you're creating a wrapper for Servo, in order to provide an UI around the browser engine, but I think I'm missing something here.  
  
BrowserHTML is an Electron application: Electron itself is a framework that includes Webkit, another browser engine.

Is the goal of this project to create a wrapper around a browser engine, by using Electron which is itself a framework that runs with Webkit (another browser engine)? Does it even make sense?

**Comment by Happy-Ferret** [comment]
I'm not a developer of BrowserHTML, but the way I've always seen it is that this project is a sandbox of sorts. Something developers, UI designers and testers alike can play with and easily implement new features/UI paradigms on without fuzzing around or muddying the main codebase.

It's probably not unlike Opera's Neon in that regard.

Hint: Just compare what it takes to add a new "about" page to browserhtml vs gecko-dev. Even with a bootstraped extension (which, as of Firefox 57, only work in system addons) it's considerably more work to do it in Firefox proper.

**Comment by jdanford** [comment]
@denysvitali I'm not sure if the README for this project has dramatically changed since you filed this issue, but here's the list of ways to run Browser.html:

1. Servo + Graphene
2. Gecko + Graphene
3. Electron

It's not accurate to say that Browser.html is an Electron application; that's simply the most convenient platform for development, and it's supposed to use Servo (instead, not additionally) in production. As you said yourself, this is essentially a UI wrapper for a browser engine, and Electron/Gecko is filling that role until Servo is more complete.

**Comment by Gozala** [comment]
Hi @denysvitali here is somewhat hand-wavy explanation:

Initially browser.html started as an experiment back in the firefox os days where we (re)used all the work that went into a firefox-os. From there browser.html went through many phases that were mostly focused at exploring different user experience user agent might provide. Fast forward from there, browser.html changed from UX experiments platform to servo nightly UI.

As of electron support, it never was primary goal of this project as mentioned we initially used gecko & later switched to servo (although it still continued to work on gecko). I can't exactly remember when this was in terms of timeline but team started get questions if they could run this on electron, I think around the time [Tofino Project](https://medium.com/project-tofino) was announced, at which point with minimal effort we made it possible to run browser.html on top of electron as well. We also kept supporting elector because:
1. We found it useful to compare embedding APIs.
2. Gecko devtools at a time was not quite as reliable as web-inspector and servo essentially had none and ability to track down some bugs using web-inspector turned out to be useful.

Hope this reasonably answers raised questions. I'll close this issue but feel free to followup.
